include/deal.II/physics/notation.h

```diff
@@ -0,0 +1,619 @@
+     *   \sqrt{2} {\cal{C}}_{0200}  & \sqrt{2} {\cal{C}}_{0211} & \sqrt{2} {\cal{C}}_{0222}  & 2 {\cal{C}}_{0212}         & 2 {\cal{C}}_{0202}         & 2 {\cal{C}}_{0201}        \\
+     *   \sqrt{2} {\cal{C}}_{0100}  & \sqrt{2} {\cal{C}}_{0111} & \sqrt{2} {\cal{C}}_{0122}  & 2 {\cal{C}}_{0112}         & 2 {\cal{C}}_{0102}         & 2 {\cal{C}}_{0101}
+     *   \end{array}\right] .
+     * @f]
+     * The mapping from the two Kelvin indices of the FullMatrix $\tilde{\cal{C}}$ to the
+     * rank-4 SymmetricTensor $\cal{C}$ can be inferred using the table shown above.
+     *
+     * An important observation is that both the left-hand side tensor $\tilde{\mathbf{S}}$
+     * and right-hand side tensor $\tilde{\mathbf{E}}$ have the same form; this is a property
+     * that is not present in Voigt notation.
+     * The various factors introduced into $\tilde{\mathbf{S}}$,  $\tilde{\mathbf{E}}$
+     * and $\tilde{\cal{C}}$ account for the symmetry of the tensors. The Kelvin
+     * description of their non-symmetric counterparts include no such factors.
+     *
+     * Some useful references to the that show how this notation works includes <a
+     * href="XXX">the unit tests</a>, as well as:
+     * @code{.bib}
+     * @Article{Nagel2016,
+     * author    = {Nagel, T. and G{\"o}rke, U-J. and Moerman, K. and Kolditz, O.},
+     * title     = {On advantages of the Kelvin mapping in finite element implementations of deformation processes},
+     * journal   = {Environmental Earth Sciences},
+     * year      = {2016},
+     * volume    = {75},
+     * number    = {11},
+     * pages     = {937}
+     * }
+     * @endcode
+     * as well as the online references found <a
+     * href="http://sepwww.stanford.edu/oldsep/joe/Reprints/8IWSA.pdf">here</a> and <a
+     * href="http://books.google.de/books?id=Ptxu63ARr7UC&pg=PA298&lpg=PA298&dq=kelvin+tensor+notation&source=bl&ots=LgKytTBYu9&sig=LHB3IrBRaJLfI8-IEDsBOQz_a4c&hl=en&sa=X&ei=1xZtUeSpBoOqtAa224DQAQ&redir_esc=y#v=onepage&q=kelvin%20tensor%20notation&f=false">here</a>.
+     *
+     * @author Jean-Paul Pelteret, 2017
+     */
+    template <int dim>
+    struct Kelvin
+    {
+      /**
+       * @name Forward operation: Tensor notation to Kelvin notation
+       */
+//@{
+
+      /**
+       * Convert a scalar value to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const Number &s);
+
+
+      /**
+       * Convert a rank-0 tensor to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const Tensor<0,dim,Number> &s);
+
+
+      /**
+       * Convert a rank-1 tensor to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const Tensor<1,dim,Number> &v);
```

Comment:
i wonder if one can void unrolling and extra copying by using `ArrayView`?  similar to https://github.com/dealii/dealii/issues/5021

Reply:
As per our discussion, this is not relevant as I want to potentially manipulate the resultant vectors and matrices.